File: apps/api/app/schemas/cockpit_alerts_needing_attention.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class CockpitAttentionSummarySchema(BaseModel):
    headline: str
    total_items: int
    high_priority: int
    medium_priority: int
    low_priority: int
    unknown_priority: int
    active_alerts: int
    unresolved_incidents: int
    monitor_degraded: int
    stale_data: int
    risk_attention: int
    trading_halt: int
    missing_context: int


class CockpitAttentionItemSchema(BaseModel):
    id: str
    source: Literal[
        "alert",
        "incident",
        "monitor",
        "risk",
        "trading_halt",
        "notification",
        "paper",
        "unknown",
    ]
    title: str
    message: str
    priority: Literal["high", "medium", "low", "unknown"]
    status: str
    detected_at: str | None
    attention_type: Literal[
        "active_alert",
        "unresolved_incident",
        "monitor_degraded",
        "stale_data",
        "risk_attention",
        "trading_halt",
        "missing_context",
    ]
    evidence: list[str]
    missing_data: list[str]
    recommended_review_action: str
    is_actionable: Literal[False]


class CockpitAttentionGroupSchema(BaseModel):
    group: str
    count: int
    item_ids: list[str]


class CockpitAlertsNeedingAttentionResponseSchema(BaseModel):
    generated_at: str
    mode: Literal["paper"]
    summary: CockpitAttentionSummarySchema
    attention_items: list[CockpitAttentionItemSchema]
    grouped_by_priority: list[CockpitAttentionGroupSchema]
    grouped_by_source: list[CockpitAttentionGroupSchema]
    monitor_notes: list[str]
    risk_notes: list[str]
    limitations: list[str]
    recommended_review_actions: list[str]

File: apps/web/lib/api/cockpitAlertsNeedingAttention.ts
import { apiRequest } from "./core";

export type CockpitAttentionPriority = "high" | "medium" | "low" | "unknown";
export type CockpitAttentionSource =
  | "alert"
  | "incident"
  | "monitor"
  | "risk"
  | "trading_halt"
  | "notification"
  | "paper"
  | "unknown";
export type CockpitAttentionType =
  | "active_alert"
  | "unresolved_incident"
  | "monitor_degraded"
  | "stale_data"
  | "risk_attention"
  | "trading_halt"
  | "missing_context";

export interface CockpitAttentionSummary {
  headline: string;
  total_items: number;
  high_priority: number;
  medium_priority: number;
  low_priority: number;
  unknown_priority: number;
  active_alerts: number;
  unresolved_incidents: number;
  monitor_degraded: number;
  stale_data: number;
  risk_attention: number;
  trading_halt: number;
  missing_context: number;
}

export interface CockpitAttentionItem {
  id: string;
  source: CockpitAttentionSource;
  title: string;
  message: string;
  priority: CockpitAttentionPriority;
  status: string;
  detected_at: string | null;
  attention_type: CockpitAttentionType;
  evidence: string[];
  missing_data: string[];
  recommended_review_action: string;
  is_actionable: false;
}

export interface CockpitAttentionGroup {
  group: string;
  count: number;
  item_ids: string[];
}

export interface CockpitAlertsNeedingAttentionResponse {
  generated_at: string;
  mode: "paper";
  summary: CockpitAttentionSummary;
  attention_items: CockpitAttentionItem[];
  grouped_by_priority: CockpitAttentionGroup[];
  grouped_by_source: CockpitAttentionGroup[];
  monitor_notes: string[];
  risk_notes: string[];
  limitations: string[];
  recommended_review_actions: string[];
}

export async function getCockpitAlertsNeedingAttention(): Promise<CockpitAlertsNeedingAttentionResponse> {
  return apiRequest<CockpitAlertsNeedingAttentionResponse>("/cockpit/alerts-needing-attention", {
    method: "GET",
  });
}
